Source/WebKit/UIProcess/win/PageClientImpl.cpp

```diff
@@ -232,6 +240,70 @@ void PageClientImpl::preferencesDidChange()
     notImplemented();
 }
 
+bool PageClientImpl::handleRunOpenPanel(WebPageProxy*, WebFrameProxy*, const FrameInfoData&, API::OpenPanelParameters* parameters, WebOpenPanelResultListenerProxy* listener)
+{
+    ASSERT(parameters);
+    ASSERT(listener);
+
+    HWND viewWindow = viewWidget();
+    if (!IsWindow(viewWindow))
+        return false;
+
+    bool isAllowMultipleFiles = parameters->allowMultipleFiles();
+    Vector<WCHAR> fileBuf(isAllowMultipleFiles ? maxFilePathsListSize : MAX_PATH);
+
+    OPENFILENAME ofn;
+
+    memset(&ofn, 0, sizeof(ofn));
+
+    // Need to zero out the first char of fileBuf so GetOpenFileName doesn't think it's an initialization string
+    fileBuf[0] = '\0';
+
+    ofn.lStructSize = sizeof(ofn);
+    ofn.hwndOwner = viewWindow;
+    String allFiles = makeString(allFilesText(), String("\0*.*\0\0", 6));
```

Comment:
Use `auto` especially if you have a word `String` in both sides.

Reply:
`auto allFiles = makeString(allFilesText(), "\0*.*\0\0"_s);`

There is no point in creating a String to call makeString(). It is more efficient to pass an ASCIILiteral.

Reply:
The results were different for String and ASCIILiteral.

> auto allFiles = makeString(allFilesText(), String("\0*.*\0\0", 6));
The length of `allFiles` is 15, and the filter field in the dialog is normal.

> auto allFiles = makeString(allFilesText(), "\0*.*\0\0"_s);
The length of `allFiles` is 9, and the filter field in the dialog is garbled.

This is just a guess, but I wonder if ASCIILiteral which contains null characters has some problem with the conversion in the middle.

Reply:
The filter string setting has been removed.
The default is the same as selecting "All Files (*.*)".